hexrd/ui/image_stack_dialog.py

```diff
@@ -326,37 +326,48 @@ def add_omega_toggled(self, checked):
     def reverse_frames(self, state):
         self.state['reverse_frames'] = state
 
+    def frames_per_image(self, num_files):
+        frames = self.state['total_frames']
+        frames -= self.state['empty_frames']
+        if self.state['max_file_frames']:
+            frames = min(frames, self.state['max_file_frames'])
+        if self.state['max_total_frames']:
+            frames = min(self.state['max_total_frames'], num_files * frames)
+            frames /= num_files
+        return frames
+
     def get_omega_values(self, num_files):
+        # Returns the omega values that are used to populate the
+        # SimpleImageSeries dialog table
+        wedges = self.state['wedges']
         if self.state['omega_from_file'] and self.state['omega']:
-            omega = np.load(self.state['omega'])
-        else:
+            # user selected a file
+            wedges = np.load(self.state['omega'])
+        if wedges and num_files > 1:
+            # we create a wedge for each image based on the number
+            # of frames per image and number of steps in each wedge
             omega = []
-            nframes = int(self.ui.total_frames.text()) // num_files
-            if self.state["wedges"]:
-                nsteps = [w[2] for w in self.state['wedges']]
-                if len(nsteps) != num_files:
-                    nsteps = [sum(nsteps) // num_files] * num_files
-                row_count = self.ui.omega_wedges.rowCount()
-                length = num_files if row_count == 1 else 1
-                for i in range(row_count):
-                    start = float(self.ui.omega_wedges.item(i, 0).text())
-                    stop = float(self.ui.omega_wedges.item(i, 1).text())
-                    steps = int(float(self.ui.omega_wedges.item(i, 2).text()))
-                    delta = (stop - start) / length
-                    omega.extend(np.linspace(
-                        [start, start + delta],
-                        [stop - delta, stop],
-                        length))
-                omega = np.array(omega)
-            else:
-                nsteps = [nframes] * num_files
-        if not len(omega):
+            for start, stop, nsteps in wedges:
+                steps = self.frames_per_image(num_files)
```

Comment:
Yes it can be. Double-checking this I realized that there was an issue with the calculation though. The `max_total_frames` was being used to adjust the number of frames per image. This is wrong though and that number should only affect the last image (essentially only removing images from the end of the imageseries). This is fixed now though.